auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue

```diff
@@ -103,269 +104,281 @@
-      'currentOrgAddress'
-    ]),
-    ...mapState(useUserStore, ['currentUser'])
-  },
-  methods: {
-    ...mapActions(useOrgStore, [
-      'validatePADInfo',
-      'getOrgPayments',
-      'updateOrg',
-      'syncAddress',
-      'setCurrentOrganizationPaymentType'
-    ])
-  }
-})
-export default class AccountPaymentMethods extends Mixins(AccountChangeMixin) {
-  private readonly setCurrentOrganizationPaymentType!: (paymentType: string) => void
-  private readonly getOrgPayments!: () => any
-  private readonly updateOrg!: (requestBody: CreateRequestBody) => Promise<Organization>
-  private readonly currentMembership!: Member
-  private readonly currentOrganization!: Organization
-  private readonly currentOrgPaymentType!: string
-  private readonly validatePADInfo!: () => Promise<PADInfoValidation>
-  private readonly permissions!: string[]
-  private readonly currentUser!: KCUserProfile
-  private readonly currentOrgAddress!: Address
-  private readonly syncAddress!: () => Address
-  private savedOrganizationType: string = ''
-  private selectedPaymentMethod: string = ''
-  private padInfo: PADInfo = {} as PADInfo
-  private isBtnSaved = false
-  private disableSaveBtn = false
-  private errorMessage: string = ''
-  private errorTitle = 'Payment Update Failed'
-  private bcolInfo: BcolProfile = {} as BcolProfile
-  private errorText = ''
-  private isLoading: boolean = false
-  private padValid: boolean = false
-  private eftValid: boolean = false
-  private paymentMethodChanged:boolean = false
-  private isFuturePaymentMethodAvailable: boolean = false // set true if in between 3 days cooling period
-  private isTOSandAcknowledgeCompleted:boolean = false // sert true if TOS already accepted
+  emits: ['emit-bcol-info'],
+  setup (props, { emit, root }) {
+    const orgStore = useOrgStore()
+    const userStore = useUserStore()
 
-  $refs: {
-      errorDialog: InstanceType<typeof ModalDialog>
-    }
+    const state = reactive({
+      savedOrganizationType: '',
+      selectedPaymentMethod: '',
+      padInfo: {} as PADInfo,
+      isBtnSaved: false,
+      disableSaveBtn: false,
+      errorMessage: '',
+      errorTitle: 'Payment Update Failed',
+      bcolInfo: {} as BcolProfile,
+      errorText: '',
+      isLoading: false,
+      padValid: false,
+      eftValid: false,
+      ejvValid: false,
+      paymentMethodChanged: false,
+      isFuturePaymentMethodAvailable: false, // set true if in between 3 days cooling period
+      isTOSandAcknowledgeCompleted: false // set true if TOS already accepted
+    })
 
-  private setSelectedPayment (payment) {
-    this.errorMessage = ''
-    this.selectedPaymentMethod = payment.selectedPaymentMethod
-    this.isBtnSaved = (this.isBtnSaved && !payment.isTouched) || false
-    this.paymentMethodChanged = payment.isTouched || false
-  }
+    const errorDialog = ref<InstanceType<typeof ModalDialog>>()
+
+    const currentOrganization = computed(() => orgStore.currentOrganization)
```

Comment:
I see this repeated alot.. might be time to create a composable to destruct into these below?

`const { currentOrganization, currentOrgAAddress, currentMembership... } = useOrgComposable()`

Reply:
added